types/index.d.ts

```diff
@@ -0,0 +1,100 @@
+declare namespace Vuex {
+  class Store<S> {
+    constructor(options: StoreOption<S>);
+
+    state: S;
+
+    dispatch(mutationName: string, ...args: any[]): void;
+    dispatch<P>(mutation: MutationObject<P>): void;
+
+    watch(path: string, cb: (value: any) => void, options?: WatchOption): void;
```

Comment:
I've merged `next` into `master` and this is the only thing that is deprecated. LGTM after removing this.